build-tools/packages/build-cli/src/args.ts

```diff
@@ -36,3 +40,38 @@ export const findPackageOrReleaseGroup = (
 		context.independentPackages.find((pkg) => pkg.nameUnscoped === name)
 	);
 };
+
+interface IFluidRepoBuildDir extends IRepoBuildDir {
```

Comment:
Without changing the existing config shape, that is very hard to do.   The only thing that might make it not as strange is to duplicate the fields here, so the only dependency is that they are "type-compatible".   And in case things change in the future, we can map the data before passing it into FluidRepo.

Other suggestions welcome.

Reply:
I think I'm open to changing the config shape somewhat.

Reply:
What is the back-compat requirement if we change it?

Reply:
I thought about this and experimented with completely splitting the fluid-build config from the flub config in #21967. For back-compat, it can still load the old, combined config. I think that it makes sense long-term for the configs to be separate, since there is very little overlap in usage outside of a few places in policy. Curious what you think about it.